### PR TITLE
cloudrunv2: Validate number of ports specified restricting to 1.

### PR DESCRIPTION
**Pull request by aebrahim**
Accidentally specifying the `containers.ports` twice results in the following error:

```
Error: Error updating Service "projects/myproject/locations/us-central1/services/myservice": googleapi: Error 400: Violation in UpdateServiceRequest.service.template.containers[0].ports: ports should contain 0 or 1 port.
│ Details:
│ [
│   {
│     "@type": "type.googleapis.com/google.rpc.BadRequest",
│     "fieldViolations": [
│       {
│         "description": "ports should contain 0 or 1 port.",
│         "field": "Violation in UpdateServiceRequest.service.template.containers[0].ports"
│       }
│     ]
│   }
│ ]
│ 
│   with google_cloud_run_v2_service.myservice,
│   on myservice.tf line 1, in resource "google_cloud_run_v2_service" "myservice":
│    1: resource "google_cloud_run_v2_service" "myservice" {
│ 
```

It would be convenient if this issue could be caught at validation time.

```release-note:enhancement
cloudrunv2: added the validation to restrict number of ports to 1 to match server-side validation
```

### CODE REVIEW

**Comment by zli82016** [comment]
Checks in this PR are running now.

**Comment by zli82016** [comment]
Hello, @aebrahim , normally we don't add client side validation in Terraform providers and rely on the server side validation, in case the service side validation will change. 

Also, it is a breaking change to add a validation for a field and should not be done in a minor release.
https://googlecloudplatform.github.io/magic-modules/develop/breaking-changes/breaking-changes/

**Comment by aebrahim** [comment]
Thanks for the response. Understood on this being a breaking change.

My team has really struggled with the lack of validation before deployment. That means it is very very hard to test terraform configurations before deploying, and we have had countless errors due to this.

If there was a way in terraform to run server-side validation before deployment, it would be a different story, but I am not aware of any. Would you be willing to revisit the decision to not use client-side validation generally?

IMHO, not being able to test the configuration before deployment for validity negates a good chunk of the value for infrastructure as code.

**Comment by zli82016** [comment]
> google_cloud_run_v2_service

I am not so familiar with the validation of the deployment. 
After adding `maxItem` to `ports`, will it do the validation before deployment? How do you test terraform configurartions before deployment?

File: mmv1/products/cloudrunv2/Service.yaml [hunk 1]
```diff
@@ -448,6 +448,7 @@ properties:
                     Determines whether CPU should be boosted on startup of a new container instance above the requested CPU threshold, this can help reduce cold-start latency.
             - !ruby/object:Api::Type::Array
               name: 'ports'
+              maxSize: 1
```
Comment (zli82016) [inline mmv1/products/cloudrunv2/Service.yaml:451]: by the way, it should be `max_size`

Comment (aebrahim) [inline mmv1/products/cloudrunv2/Service.yaml:451]:
```suggestion
              max_size: 1
```

Comment (aebrahim) [reply mmv1/products/cloudrunv2/Service.yaml]: done!

**Comment by aebrahim** [comment]
Thank you @zli82016 

We run `terraform validate` as a part of our pre-submission checks before merging code into our repo. Otherwise, `terraform apply` needs to be manually debugged with a live configuration.

**Comment by zli82016** [comment]
> 
> We run `terraform validate` as a part of our pre-submission checks before merging code into our repo. Otherwise, `terraform apply` needs to be manually debugged with a live configuration.


@melinath , the contributor wants to add `max_size` to the array `ports`. Will `max_size` work with `terraform validate`? The server side already has the similar validation. Based on my understanding, the contributor wants to catch the error earlier.

**Comment by melinath** [comment]
Based on https://developer.hashicorp.com/terraform/cli/commands/validate I don't think that validate runs `ValidateFuncs` but I could be wrong. They would definitely run as part of `plan` though. Either way - adding `max_size` is a breaking change unless configurations with more elements are invalid (that is, the API would return an error.) If this limit is already enforced on the API side and we're just making it clear in TF then the change should be fine in a minor release.

**Comment by melinath** [comment]
@zli82016 please double-check the API behavior.

**Comment by zli82016** [comment]
> @zli82016 please double-check the API behavior.

Confirmed that the API returns an error with more than one ports.

```
error=
  | Error running apply: exit status 1
  | 
  | Error: Error creating Service: googleapi: Error 400: Violation in CreateServiceRequest.service.template.containers[0].ports: ports should contain 0 or 1 port.
  | Details:
  | [
  |   {
  |     "@type": "type.googleapis.com/google.rpc.BadRequest",
  |     "fieldViolations": [
  |       {
  |         "description": "ports should contain 0 or 1 port.",
  |         "field": "Violation in CreateServiceRequest.service.template.containers[0].ports"
  |       }
  |     ]
  |   },
  |   {
  |     "@type": "type.googleapis.com/google.rpc.DebugInfo",
  |     "detail": "[ORIGINAL ERROR] generic::invalid_argument: userFacingMessage: Violation in CreateServiceRequest.service.template.containers[0].ports: ports should contain 0 or 1 port.; userFacingDetails: field_violations {\n  field: \"Violation in CreateServiceRequest.service.template.containers[0].ports\"\n  description: \"ports should contain 0 or 1 port.\"\n}\n; \ncom.google.cloud.eventprocessing.manager.api.error.MalformedRequestException: userFacingMessage: Violation in CreateServiceRequest.service.template.containers[0].ports: ports should contain 0 or 1 port.; userFacingDetails: field_violations {\n  field: \"Violation in CreateServiceRequest.service.template.containers[0].ports\"\n  description: \"ports should contain 0 or 1 port.\"\n}\n;  Code: INVALID_ARGUMENT [google.rpc.error_details_ext] { message: \"Violation in CreateServiceRequest.service.template.containers[0].ports: ports should contain 0 or 1 port.\" details { [type.googleapis.com/google.rpc.BadRequest] { field_violations { field: \"Violation in CreateServiceRequest.service.template.containers[0].ports\" description: \"ports should contain 0 or 1 port.\" } } } }"
  |   }
  | ]
  | 
  |   with google_cloud_run_v2_service.default,
  |   on terraform_plugin_test.tf line 2, in resource "google_cloud_run_v2_service" "default":
  |    2: resource "google_cloud_run_v2_service" "default" {
```

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**